src/adapter.js

```diff
@@ -12,7 +12,51 @@ export const mediaTypes = [
 ];
 
 export function detect(source) {
-  return !!source.match(/"?swagger"?:\s*["']2\.0["']/g);
+  return !!(_.isString(source)
+    ? source.match(/"?swagger"?:\s*["']2\.0["']/g)
+    : source.swagger === '2.0');
+}
+
+// Test whether a key is a special Swagger extension.
+function isExtension(value, key) {
+  return key.indexOf('x-') === 0;
+}
+
+// Test whether tags can be treated as resource groups, and if so it sets a
+// group name for each resource (used later to create groups).
```

Comment:
I believe this goes back to your comment on how we handle resource groups in Refract. Seems like the best solution is to go with what you have, like you've said, but think through things in the API Description spec itself to have better support for tags.

Thoughts?